core/src/main/scala/org/apache/spark/sql/execution/columnar/ColumnTableScan.scala

```diff
@@ -426,37 +492,75 @@ private[sql] final case class ColumnTableScan(
           ($wasNullClass)null);
       """
     }
+    val commonSnippet =
+      s"""
+         $input = $colInput;
+         $inputIsRow = false;
+         $inputIsRowSRR = false;
+         if ($input == null || !$input.hasNext()) {
+          return false;
+         }
+       """.stripMargin
     val nextBatch = ctx.freshName("nextBatch")
     ctx.addNewFunction(nextBatch,
       s"""
-         |private boolean $nextBatch() throws Exception {
-         |  if ($buffers != null) return true;
-         |  // get next batch or row (latter for non-batch source iteration)
-         |  if ($input == null) return false;
-         |  if (!$input.hasNext()) {
-         |    if ($input == $rowInput) {
-         |      $input = $colInput;
-         |      $inputIsRow = false;
-         |      if ($input == null || !$input.hasNext()) {
-         |        return false;
-         |      }
-         |    } else {
-         |      return false;
-         |    }
-         |  }
-         |  if ($inputIsRow) {
-         |    $nextRowSnippet
-         |    $numBatchRows = 1;
-         |  } else {
-         |    $batchInit
-         |    $numOutputRows.add($numBatchRows);
-         |    // initialize the column buffers and decoders
-         |    ${columnBufferInitCode.toString()}
-         |  }
-         |  $batchIndex = 0;
-         |  return true;
-         |}
-      """.stripMargin)
+         private boolean $nextBatch() throws Exception {
+           if ($buffers != null) return true;
+           // get next batch or row (latter for non-batch source iteration)
+           if ($input == null) return false;
+           if (!$input.hasNext()) {
+             if ($input == $rowInput) {
+      """ +
+             (if (isForSampleReservoirAsRegion) {
+                 s"""
+                     $input = $rowInputSRR;
+                     $inputIsRow = false;
+                     $inputIsRowSRR = true;
+                 """.stripMargin +
+                    output.zipWithIndex.map{
+                     case (attr, index) =>
+                       val baseIndex = baseRelation.schema.fieldIndex(attr.name)
+                       s"""${variableBuffer(index)} = new $rowAdapterClass($unsafeHolder,
+                          $baseIndex);""".stripMargin
+                   }.mkString("\n") +
+                 s"""
+                   if($input == null  || !$input.hasNext()) {
+                     $commonSnippet
+                   }
+                  """.stripMargin
+              } else {
+                commonSnippet
+              }
+             ) +
+        s"""}""" +
+        (if (isForSampleReservoirAsRegion) {
+         s"""
+            else if($input == $rowInputSRR) {
+              $commonSnippet
+            }
+          """.stripMargin
+        } else {
+          ""
+        }) +
```

Comment:
@ahshahid
WTH IS THIS CODE READABILITY AND FORMATTING???? You have completely imploded and destroyed the original code. Is it too difficult to declare separately variables? Completely unacceptable.